### PR TITLE
Support for Microsoft Edge from Windows build 15002 and up

### LINKED ISSUE #6713: Windows 10 build 15002: cannot activate browse/focus mode in Edge, cannot navigate to the document by tabbing around
Hi,

Related to #6712: In Windows 10 build 15002, when using Edge, one cannot move to the document when tabbing around nor toggle focus/browse mode.

STR:

1. Install build 15002.
2. Open Edge, then try tabbing around to see if you can locate the document.

Expected: one can tab to the document and toggle focus/browse mode.
Actual: Where there should have been a document, NVDA says, "unknown".
Observation: in build 15002, there is a new "frame" that is sort of a parent of the actual document, whereas this isn't there in prior builds.
Impact: Major, as this prevents people from testing various enhancements to Edge support that are now part of master branch snapshots.

Setup:

* NVDA version: next.13795
* OS: Windows 10 build 15002

Thanks.

### PR DESCRIPTION
**Pull request by michaelDCurran**
Handle the change to the Edge document container UIAElement, and explicitly allow events for Windows.UI.Core windows as foreground checks will no longer work for UWP. 
Fixes #6713

### CODE REVIEW

**Comment by michaelDCurran** [comment]
it is possible that the container change will be fixed by Microsoft before RS2. However the foreground descendant change is deliberate going forward.

**Review by jcsteh — APPROVED** [review]

**Comment by gdata1** [comment]
Hello. on some sites such as YouTube NVDA does not see
some of the buttons when using the arrow keys.try doing the following. 1. open microsoft edge
2. go to the website of yutube.com
3. use the navigation buttons by pressing B shift + B nvda next-13843,f979ca07
 windows 10 15019

**Comment by michaelDCurran** [comment]
@jcsteh:  it will be difficult to check for microsoftedgecp in shouldAcceptEvent as we need to some how get an appmodule from a windowhandle. However, for now instead I have tightened up the  check for Windows.UI.Core in general, by also checking that the window is equal to or a descendant of the active window in the current input thread. This works nicely for Edge.

File: source/eventHandler.py [hunk 1]
```diff
@@ -256,8 +256,10 @@ def shouldAcceptEvent(eventName, windowHandle=None):
 		# #5595: Events for the cursor get mapped to the desktop window.
 		return True
 
-	# #6713: Edge (and soon all UWP apps) will no longer have windows as descendants of the foreground window
-	if wClass.startswith('Windows.UI.Core'):
+	# #6713: Edge (and soon all UWP apps) will no longer have windows as descendants of the foreground window.
+	# However, it does look like they are always  equal to or descendants of the "active" window of the input thread. 
+	gi=winUser.getGUIThreadInfo(0)
+	if winUser.isDescendantWindow(gi.hwndActive,windowHandle) and wClass.startswith('Windows.UI.Core'):
 		return True
```
Comment (jcsteh) [inline source/eventHandler.py]: We don't need GUIThreadInfo outside this yet, so I think I'd prefer this inside the condition:

```
if wClass.startswith('Windows.UI.Core'):
	return isDescendant blahblah
```

**Review by jcsteh — APPROVED** [review]

**Comment by jcsteh** [comment]
We're going to incubate this just for a few days and then merge to master early. We really want this in 2017.1 and it's thus important that it gets wider beta testing.

**Comment by PratikP1** [comment]
Let us know when you want testing with Edge to begin. I’ve been keeping my hands off edge and NVDA.

**Comment by michaelDCurran** [comment]
@jcsteh: Along with the fix you requested in the last review, I also fixed the one and only issue that occurs when this code is run with the Edge 'gap' removed: namely an endless recursion in Edge's _getTextwithFieldsFromUIARange. Commit 1b20ccd is a tiny change that just checks for a collapsed range and if so returns, and probably should have always been there anyway.  You should review it, however I took the choice of merging it to Next along with the other change as time is quickly running out.

**Review by jcsteh — APPROVED** [review]

**Comment by gdata1** [comment]
Hello. When you switch keyboard layouts nvda continues to refer to the symbols on the keyboard recruited which was originally. helps you switch between Windows.
windows 10 15025

**Comment by gdata1** [comment]
Hello. When you switch keyboard layouts in microsoft edge nvda continues to refer to the symbols on the keyboard recruited which was originally. helps you switch between Windows.
windows 10 15025

**Comment by gdata1** [comment]
Here's what the error I see when viewing the log.
[nvda.txt](https://github.com/nvaccess/nvda/files/759711/nvda.txt)

**Comment by josephsl** [comment]
Hi, covered by #6712, please post your comments there. Thanks.

**Pull request merged**